whois-query/src/main/java/net/ripe/db/whois/query/planner/AbuseCFinder.java

```diff
@@ -1,5 +1,6 @@
 package net.ripe.db.whois.query.planner;
 
+import com.google.common.collect.ImmutableSet;
```

Comment:
I suggest using the built-in wrapper java.util.Collections.unmodifiableSet(<Set>) instead of guava

Reply:
Collections.unmodifiableSet() still can be changed as its a view that's why i usually prefer immutable set

Reply:
A view is fine, if you don't have any other references to the actual set (i.e. wrap creating the Set in a method, and return the view). We shouldn't have multiple ways to do the same thing in the code. But here it's not a big deal.